src/components/TextField/TextField.tsx

```diff
@@ -15,17 +17,34 @@ function getRandomDigits() {
 
 export const TextField: React.FC<Props> = ({
   name,
-  value,
   label = name,
   required = false,
-  onChange = () => {},
+  approveField,
+  editMovie,
 }) => {
-  // generage a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
 
-  // To show errors only if the field was touched (onBlur)
-  const [touched, setToched] = useState(false);
-  const hasError = touched && required && !value;
+  const [touched, setTouched] = useState(false);
+
+  const [value, setValue] = useState(initValue);
+
+  const hasErrors = () => {
```

Comment:
you can add `useMemo` here if you want. `hasErrors` won't return a function but will calculate the value instead 
```suggestion
  const hasErrors = useMemo(() => {
```